packages/notification-center/src/store/styles/styles-provider.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { FunctionComponent, createContext, useMemo, useContext } from 'react';
+import { CSSInterpolation } from '@emotion/css';
+import get from 'lodash.get';
+
+import { NotificationCenterStyles, StylesPaths } from './styles-provider.types';
+import { CSSFunctionOrObject } from '../../components/types';
+import { useNovuTheme } from '../../hooks';
+
+const StylesContext = createContext<{ styles: NotificationCenterStyles } | undefined>(undefined);
+
+export const useStyles = (path: StylesPaths | StylesPaths[]): CSSInterpolation[] => {
+  const stylesContext = useContext(StylesContext);
+  const { theme, colorScheme, common } = useNovuTheme();
+
+  if (!stylesContext) {
+    throw new Error('useStyles must be used within a StylesProvider');
+  }
+
+  const getStyleByPath = (pathInStyles: StylesPaths): CSSInterpolation => {
+    const stylePart: CSSFunctionOrObject = get(stylesContext.styles, pathInStyles);
+
+    return typeof stylePart === 'function' ? stylePart({ theme, common, colorScheme }) : stylePart;
+  };
+
+  if (Array.isArray(path)) {
+    return path.map((el) => getStyleByPath(el));
+  }
+
+  return [getStyleByPath(path)];
+};
```

Comment:
the styles hook: it takes the styles path (or array) which is a path to objects that are holding the styles, and it uses that path to take a object style which is then returned... if that path returns the function then the function is called with args and it must return the styles object... 
the paths are defined by a union of strings...